processing/src/main/java/io/druid/query/BySegmentQueryRunner.java

```diff
@@ -61,7 +62,11 @@ public BySegmentQueryRunner(
                   new BySegmentResultValueClass<T>(
                       results,
                       segmentIdentifier,
-                      query.getIntervals().get(0)
+                      Iterables.getOnlyElement(
+                          Iterables.getOnlyElement(query.getDataSources())
```

Comment:
Would you tell me why you think so? I think every query should be regarded to have one or more data sources basically.